### ISSUE TITLE
Installation fails on PHP 5.6.35 and MySQL 5.7.21

### ISSUE THREAD

**Issue by a20eac1d** [ABXD/ABXD#37]
`
Setting utf8_unicode_ci collation to the database...
Installing database version 301...
badges… OK.
settings… creating…\abxd\lib\mysql.php:164   backTrace()
\abxd\lib\mysql.php:131   rawQuery('...')
\abxd\lib\mysqlfunctions.php:71   query('...')
\abxd\install\installer.php:72   Upgrade()
\abxd\install\doinstall.php:7   install()

BLOB, TEXT, GEOMETRY or JSON column 'value' can't have a default value

Query was: create table `settings` (
	`plugin` varchar(128) NOT NULL DEFAULT '',
	`name` varchar(128) NOT NULL DEFAULT '',
	`value` text DEFAULT '',
	unique key `mainkey` (`plugin`,`name`)
) ENGINE=MyISAM;MySQL Error.

An error occured. Please go back and fix the settings and try again
`

Apache/2.4.33 (Win64)
PHP Version 5.6.35
MySQL 5.7.21

**Comment by phase** [comment]
I fixed that here: https://github.com/phase/Blargboard/blob/a4bbdbf048792c0e348e4157f20cae1f4b6c07ee/lib/mysql.php#L13-L14

**Comment by a20eac1d** [comment]
And how do I fix this issue? I can't find blargboard in the folder.

**Comment by phase** [comment]
You need to put those lines in yourself. This obviously isn't Blargboard. Put it in https://github.com/ABXD/ABXD/blob/master/lib/mysql.php

**Comment by NightYoshi370** [comment]
Would be good if this can properly be fixed instead of downgrading MySQL. Though, I doubt that'll happen any time soon

**Comment by Windows98FE** [comment]
@NightYoshi370 Indeed. ABXD is dead and hasn't been used in a long time for anything that has been substantial. Therefore developers have absolutely no need to bother with its development anymore...

**Comment by NightYoshi370** [comment]
It hasn't been used for a long time in general. Last board I remember made using it was in 2016 where Jamie made his 9000'th board. Kind of sad, because this software had potential

**Comment by MoonlightCapital** [comment]
@Windows98FE, it's called freedom of choice, and tbh, I like ABXD as well.

**Comment by Windows98FE** [comment]
@NightYoshi370 Derpboards made by well known idiots don't count.

**Comment by Dirbaio** [comment]
There's https://nsmbhd.net running a forked version of ABXD, still somewhat maintained (runs well on PHP 7 and MySQL 5.7 with strict mode enabled). It has lots of NSMBHD-specific stuff though. 

Sadly, I don't have the time to maintain the two forks, so vanilla ABXD is mostly unmaintained now.

**Comment by laffuchino** [comment]
MariaDB 10.4 works fine with my schema edits (I PR'd them here already if you want to look).

MySQL 5.7 probably doesn't work with strict mode on which could prove problematic on shared hosting environments, then again most places have replaced MySQL with MariaDB by now...

In regards to the project being pointless working on, it's more or less a case of ABXD (and forums in general) having a niche/association with seasoned Internet users - and "socially awkward" individuals - back in the day, which largely got replaced with Discord and Twitter for most people. I guess quite a bit of the tarnishing in ways could also come from my own retardation but Discord/Twitter as opposed to forums is where the socially awkward teens of now go.
